CLI tests fail locally when the shell exports FORCE_COLOR (23 spurious failures)

## Summary

23 CLI tests fail locally whenever the ambient shell exports `FORCE_COLOR` (Claude Code, Codex, and several terminal setups do). Rich's module-level `Console()` objects autodetect from the process environment rather than from the stream `typer.testing.CliRunner` hands them, so captured output comes back full of ANSI SGR escapes and every substring assertion misses.

CI is green because GitHub Actions does not set `FORCE_COLOR` — so this is pure local noise, but it is persistent noise: every agent or contributor with a colour-forcing shell sees 23 red tests on an untouched `main` and has to re-derive that they are spurious. It came up again while validating #1659.

## Reproduce

On unmodified `main`:

```bash
FORCE_COLOR=3 uv run pytest -m "not requires_matrix" \
  tests/test_cli_config.py tests/test_cli_doctor.py tests/test_cli_trigger.py \
  tests/test_plugin_check.py tests/test_plugin_install.py
# 23 failed, 215 passed
```

Two failure shapes, same cause:

```
AssertionError: assert '7 passed' in '\x1b[1mMindRoom Doctor\x1b[0m\n\n\x1b...
yaml.reader.ReaderError: unacceptable character #x001b: special characters are not allowed
```

The second is `mindroom config init --print` output being parsed as YAML with escapes embedded.

## Root cause

`FORCE_COLOR` is set in the environment. Rich honours it and forces a colour terminal even though `CliRunner` is capturing to a non-tty buffer. The consoles are constructed at import time, so nothing per-test can influence them:

- `src/mindroom/cli/config.py:54` — `console = Console()`
- `src/mindroom/cli/desktop.py:25-26` — `_console = Console()`, `_error_console = Console(stderr=True)`
- `src/mindroom/cli/service.py:17-18` — `_console = Console()`, `_err_console = Console(stderr=True)`

Note `FORCE_COLOR=` (empty) does **not** help — Rich treats mere presence of the variable as force-on.

## Workaround

`TERM=dumb` suppresses it completely:

```bash
TERM=dumb uv run pytest -m "not requires_matrix"   # 11658 passed, 41 skipped
```

`NO_COLOR=1` alone is **not** enough: it strips colour but leaves bold/other SGR codes, so 3 of the 23 still fail.

## Suggested fix

An autouse fixture in `tests/conftest.py` that neutralises the colour-forcing variables (`FORCE_COLOR`, `CLICOLOR_FORCE`, `COLORTERM`) and pins `TERM=dumb` for the test session via `monkeypatch.delenv` / `setenv`, so the suite is deterministic regardless of the invoking shell.

That alone may not be sufficient given the consoles are module-level and may already be imported before the fixture runs — if so, the alternative is to build the `Console()` objects lazily (or accept an injected console), which also makes them testable directly. Either way the goal is that a bare `uv run pytest` is green in any shell.

## Affected tests

<details>
<summary>23 tests</summary>

```
tests/test_cli_config.py::TestRunErrorHandling::test_run_missing_config
tests/test_cli_config.py::TestConfigInit::test_init_print_outputs_config_without_writing_files
tests/test_cli_config.py::TestConfigInit::test_init_print_defaults_to_openai_without_prompting
tests/test_cli_config.py::TestConfigInit::test_init_print_ignores_existing_config_without_prompting
tests/test_cli_config.py::TestConfigInit::test_init_print_preserves_existing_env_and_storage_without_prompting
tests/test_cli_config.py::TestDoctor::test_all_checks_pass
tests/test_cli_config.py::TestDoctor::test_warns_when_encryption_store_missing
tests/test_cli_config.py::TestDoctor::test_missing_api_key_is_warning
tests/test_cli_config.py::TestDoctor::test_provider_summary_multiple_providers
tests/test_cli_config.py::TestDoctor::test_vertexai_claude_connection_check_passes
tests/test_cli_config.py::TestDoctor::test_vertexai_claude_connection_checks_each_configured_model
tests/test_cli_config.py::TestDoctor::test_vertexai_claude_api_rejection_is_failure
tests/test_cli_config.py::TestDoctor::test_memory_llm_missing_key_is_warning
tests/test_cli_config.py::TestConnect::test_connect_no_persist_prints_exports
tests/test_cli_config.py::TestConnect::test_connect_uses_runtime_env_default_provisioning_url
tests/test_cli_config.py::TestLocalStackSetup::test_no_persist_env_prints_inline_command
tests/test_cli_doctor.py::test_doctor_counts_credential_sync_value_error_as_warning
tests/test_cli_trigger.py::test_trigger_keygen_writes_private_key_and_prints_public_key
tests/test_cli_trigger.py::test_trigger_send_builds_default_signed_request
tests/test_cli_trigger.py::test_trigger_send_reports_status_error
tests/test_cli_trigger.py::test_trigger_send_escapes_status_error_markup
tests/test_plugin_check.py::test_plugins_check_cli_reports_compatibility
tests/test_plugin_install.py::test_plugins_install_cli_reports_result_and_config_snippet
```

</details>

Environment: macOS arm64, CPython 3.13.10, uv 0.11.31, shell with `FORCE_COLOR=3`, `TERM=xterm-256color`, `COLORTERM=truecolor`.

--- a/conftest.py
+++ b/conftest.py
@@ -4,20 +4,19 @@ This file exists at the repository root so pytest imports it before
 ``tests/conftest.py`` and before any test module imports the MindRoom CLI.
 """
 
-from __future__ import annotations
-
 import os
 
-# Rich resolves colour support once, when a `Console` is constructed, and the CLI builds
-# its consoles at import time (`mindroom.cli.config`, `.desktop`, `.service`). Typer does
-# the same for its own error consoles. A shell that forces colour (Claude Code, Codex and
-# several terminals export `FORCE_COLOR`) therefore bakes ANSI escapes into output that
-# `typer.testing.CliRunner` captures from a non-tty buffer, breaking substring assertions
-# and YAML parsing of `mindroom config init --print`.
+# Rich resolves color support once, when a `Console` is constructed, and the CLI builds
+# its consoles at import time (`mindroom.cli.config`, `.desktop`, `.service`), so no
+# fixture can influence them. A shell that exports `FORCE_COLOR` (Claude Code, Codex and
+# several terminals do) makes Rich force a color terminal even though
+# `typer.testing.CliRunner` captures to a non-tty buffer, baking ANSI escapes into output
+# that tests assert on and parse as YAML.
 #
-# Neutralise the forcing variables and pin a dumb terminal here, before anything imports
-# Rich or Typer, so `pytest` behaves identically in every shell. `NO_COLOR` alone is not
-# enough: it strips colour but leaves bold and other SGR codes.
-for _colour_forcing_var in ("CLICOLOR_FORCE", "COLORTERM", "FORCE_COLOR", "PY_COLORS"):
-    os.environ.pop(_colour_forcing_var, None)
+# Pinning a dumb terminal is the one switch that settles it, because Rich gates all three
+# sources of noise on `Console.is_dumb_terminal`: color (`_detect_color_system` returns
+# `None`), control codes (`Console.control` writes nothing) and width (`Console.size`
+# returns a fixed 80x25 instead of the invoking terminal's size). It wins over
+# `FORCE_COLOR` and over Typer's own `force_terminal` consoles. Setting `NO_COLOR`
+# instead would not do: it strips color but leaves bold and other SGR codes.
 os.environ["TERM"] = "dumb"

--- a/conftest.py
+++ b/conftest.py
@@ -1,0 +1,23 @@
+"""Root pytest configuration.
+
+This file exists at the repository root so pytest imports it before
+``tests/conftest.py`` and before any test module imports the MindRoom CLI.
+"""
+
+from __future__ import annotations
+
+import os
+
+# Rich resolves colour support once, when a `Console` is constructed, and the CLI builds
+# its consoles at import time (`mindroom.cli.config`, `.desktop`, `.service`). Typer does
+# the same for its own error consoles. A shell that forces colour (Claude Code, Codex and
+# several terminals export `FORCE_COLOR`) therefore bakes ANSI escapes into output that
+# `typer.testing.CliRunner` captures from a non-tty buffer, breaking substring assertions
+# and YAML parsing of `mindroom config init --print`.
+#
+# Neutralise the forcing variables and pin a dumb terminal here, before anything imports
+# Rich or Typer, so `pytest` behaves identically in every shell. `NO_COLOR` alone is not
+# enough: it strips colour but leaves bold and other SGR codes.
+for _colour_forcing_var in ("CLICOLOR_FORCE", "COLORTERM", "FORCE_COLOR", "PY_COLORS"):
+    os.environ.pop(_colour_forcing_var, None)
+os.environ["TERM"] = "dumb"

--- a/tests/test_cli_config.py
+++ b/tests/test_cli_config.py
@@ -55,6 +55,15 @@ if TYPE_CHECKING:
 runner = CliRunner()
 
 
+def test_cli_console_renders_without_ansi_escapes(capsys: pytest.CaptureFixture[str]) -> None:
+    """The root conftest must keep the import-time CLI console colourless in any shell."""
+    config_cli.console.print("[bold red]MindRoom Doctor[/bold red]")
+
+    captured = capsys.readouterr().out
+    assert "\x1b" not in captured
+    assert "MindRoom Doctor" in captured
+
+
 @pytest.fixture(autouse=True)
 def _clear_runtime_path_env(monkeypatch: pytest.MonkeyPatch) -> None:
     monkeypatch.delenv("MINDROOM_CONFIG_PATH", raising=False)

--- a/tests/test_cli_config.py
+++ b/tests/test_cli_config.py
@@ -55,15 +55,6 @@ if TYPE_CHECKING:
 runner = CliRunner()
 
 
-def test_cli_console_renders_without_ansi_escapes(capsys: pytest.CaptureFixture[str]) -> None:
-    """The root conftest must keep the import-time CLI console colourless in any shell."""
-    config_cli.console.print("[bold red]MindRoom Doctor[/bold red]")
-
-    captured = capsys.readouterr().out
-    assert "\x1b" not in captured
-    assert "MindRoom Doctor" in captured
-
-
 @pytest.fixture(autouse=True)
 def _clear_runtime_path_env(monkeypatch: pytest.MonkeyPatch) -> None:
     monkeypatch.delenv("MINDROOM_CONFIG_PATH", raising=False)
@@ -100,6 +91,15 @@ def _write_minimal_runtime_config(path: Path) -> None:
         "authorization:\n  global_users: []\n",
         encoding="utf-8",
     )
+
+
+def test_cli_console_renders_without_ansi_escapes(capsys: pytest.CaptureFixture[str]) -> None:
+    """The root conftest must keep the import-time CLI console plain in any shell."""
+    config_cli.console.print("[bold red]MindRoom Doctor[/bold red]")
+
+    captured = capsys.readouterr().out
+    assert "\x1b" not in captured
+    assert "MindRoom Doctor" in captured
 
 
 def test_cli_import_keeps_help_path_runtime_modules_lazy() -> None:

--- a/tests/test_cli_trigger.py
+++ b/tests/test_cli_trigger.py
@@ -10,7 +10,6 @@ from typing import cast
 
 import httpx
 import pytest
-from click.utils import strip_ansi
 from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PrivateKey
 from cryptography.hazmat.primitives.serialization import Encoding, NoEncryption, PrivateFormat, PublicFormat
 from typer.testing import CliRunner
@@ -478,8 +477,7 @@ def test_trigger_send_rejects_data_json_that_is_not_object(tmp_path: Path) -> No
             "--data-json",
             "[]",
         ],
-        env={"FORCE_COLOR": "1"},
     )
 
     assert result.exit_code == 2
-    assert "--data-json must decode to a JSON object" in strip_ansi(result.output)
+    assert "--data-json must decode to a JSON object" in result.output

--- a/tests/test_services.py
+++ b/tests/test_services.py
@@ -34,7 +34,7 @@ from mindroom.services.systemd import _restart_service as _restart_systemd_servi
 from mindroom.services.systemd import _start_service as _start_systemd_service
 from mindroom.services.systemd import _stop_service as _stop_systemd_service
 
-runner = CliRunner(env={"NO_COLOR": "1", "TERM": "dumb"})
+runner = CliRunner()
 
 
 def test_build_service_command_runs_mindroom_with_uv_tool(tmp_path: Path) -> None:

